extensions/azurePublish/src/components/ResourceGroupPicker.tsx

```diff
@@ -0,0 +1,202 @@
+
+const getInfoIconStyle = (required) => {
+  return {
+    root: {
+      selectors: {
+        '&::before': {
+          content: required ? " '*'" : '',
+          color: FluentTheme.palette.red,
+          paddingRight: 3,
+        },
+      },
+    },
+  };
+};
+
+const CREATE_NEW_KEY = 'CREATE_NEW';
+
+const createNewOption: IDropdownOption = {
+  key: CREATE_NEW_KEY,
+  data: { icon: 'Add' },
+  text: formatMessage('Create new'),
+};
+
+type ResourceGroupItemChoice = {
+  name: string;
+  isNew: boolean;
+  errorMessage?: string;
+};
+
+type Props = {
+  /**
+   * The resource groups to choose from.
+   * Set to undefined to disable this picker.
+   */
+  resourceGroupNames?: string[];
+  /**
+   * The selected name of the existing resource.
+   * When undefined, the 'Create new' option will be selected.
+   */
+  selectedName?: string;
+  /**
+   * The name chosen for a new resource group.
+   * Used when the 'Create new' option is selected.
+   */
+  newName?: string;
+  /**
+   * Called when the selection or new resource name changes.
+   */
+  onChange: (item: ResourceGroupItemChoice) => void;
+};
+
+const onRenderLabel = (props) => {
+  return (
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'row',
+        marginBottom: '5px',
+      }}
+    >
+      <div
+        style={{
+          marginRight: '5px',
+          fontWeight: 600,
+          fontSize: '14px',
+        }}
+      >
+        {' '}
+        {props.label}{' '}
+      </div>
+      <TooltipHost content={props.ariaLabel}>
+        <Icon iconName="Info" styles={getInfoIconStyle(props.required)} />
+      </TooltipHost>
+    </div>
+  );
+};
+
+export const ResourceGroupPicker = ({
+  resourceGroupNames,
+  selectedName: controlledSelectedName,
+  newName: controlledNewName,
+  onChange,
+}: Props) => {
+  // ----- Hooks -----//
+  const [selectedName, setSelectedName] = React.useState(controlledSelectedName || CREATE_NEW_KEY);
+  const [newName, setNewName] = React.useState(controlledNewName || '');
+  const debouncedNewName = useDebounce<string | undefined>(newName, 300);
+  const [newNameErrorMessage, setNewNameErrorMessage] = React.useState('');
+
+  React.useEffect(() => {
+    setSelectedName(controlledSelectedName || CREATE_NEW_KEY);
+  }, [controlledSelectedName]);
+
+  React.useEffect(() => {
+    setNewName(controlledNewName || '');
+  }, [controlledNewName]);
+
+  React.useEffect(() => {
+    const alreadyExists = resourceGroupNames?.some((name) => name === debouncedNewName);
+
+    if (debouncedNewName && !debouncedNewName.match(/^[-\w._()]+$/)) {
+      setNewNameErrorMessage(
+        formatMessage(
+          'Resource group names only allow alphanumeric characters, periods, underscores, hyphens and parenthesis and cannot end in a period.'
+        )
+      );
+    } else if (alreadyExists) {
+      setNewNameErrorMessage(formatMessage('The resource already exists.'));
+    } else {
+      setNewNameErrorMessage('');
```

Comment:
consider null vs emoty string

Reply:
I prefer undefined.  Any worries with undefined?  The original code was using empty string and that made me wonder if textfield behaved correctly with null or undefined so I left it alone.